component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -357,11 +357,22 @@ private boolean validateNotBeforeClaim(long currentTimeInMillis, long timeStampS
         return true;
     }
 
-    private boolean isValidSignature(SignedJWT signedJWT, String tenantDomain,
+    private boolean isValidSignature(String clientId, SignedJWT signedJWT, String tenantDomain,
                                      String alias) throws OAuthClientAuthnException {
 
+        X509Certificate cert;
+        try {
+            cert = (X509Certificate) OAuth2Util.getX509CertOfOAuthApp(clientId, tenantDomain);
+        } catch (IdentityOAuth2Exception e) {
+            // If certificate is not configured in service provider, it will through an error.
+            // For the existing clients need to handle that error and get from truststore.
+            if (log.isDebugEnabled()) {
+                log.debug(e.getMessage(), e);
```

Comment:
shall we also mention in the log that we are falling back to truststore due to the error